go/vt/mysqlctl/mysqld.go

```diff
@@ -1070,6 +1070,7 @@ func (mysqld *Mysqld) executeMysqlScript(connParams *mysql.ConnParams, sql io.Re
 	args := []string{
 		"--defaults-extra-file=" + cnf,
 		"--batch",
+		"--default-character-set=utf8mb4",
```

Comment:
I also find this usage of shelling out to use the mysql command-line client quite strange, but assuming there were valid reasons and since we were already doing so I'm fine with this fix for now. We should re-examine this usage in the future.

Reply:
Agree that this is _weird_.
There is an old PR from @shlomi-noach to remove the `PreflightSchemaChange` code. That cleanup will make it easier to refactor this.

Reply:
What happens if someone is using `ApplySchema` to do direct DDLs and they have some other character set defined in their config file? Can this break their usage?

Reply:
> What happens if someone is using `ApplySchema` to do direct DDLs and they have some other character set defined in their config file? Can this break their usage?

In this context isn't `utf8mb4` a superset of other character sets, so we should get a one-to-one mapping and we shouldn't see any data loss in the app?

Reply:
> In this context isn't `utf8mb4` a superset of other character sets, so we should get a one-to-one mapping and we shouldn't see any data loss in the app?

If you are sure of that 👍

Reply:
> 
> `skip-preflight` no longer exists: #10716

I was confusing it with #10717. Which btw, we should review and merge 😬

Reply:
Ah yes, https://github.com/vitessio/vitess/pull/10717 still has some build issues! I'll revisit it.